test/integration/networkobservability/hubble_test.go

```diff
@@ -25,13 +27,190 @@ const (
 )
 
 var (
-	defaultRetrier  = retry.Retrier{Attempts: retryAttempts, Delay: retryDelay}
-	requiredMetrics = []string{
-		"hubble_flows_processed_total",
-		"hubble_tcp_flags_total",
-	}
+	defaultRetrier = retry.Retrier{Attempts: retryAttempts, Delay: retryDelay}
+	hubbleKeys     = make(map[string][]string)
 )
 
+func init() {
+	dns := []string{"query"}
+	flow := []string{"source", "destination"}
+	drop := []string{"source", "source"}
+
+	hubbleKeys["hubble_flows_processed_total"] = flow
+	hubbleKeys["hubble_tcp_flags_total"] = flow
+	hubbleKeys["hubble_dns_responses_total"] = dns
+	hubbleKeys["hubble_dns_response_types_total"] = dns
+	hubbleKeys["hubble_dns_queries_total"] = dns
+	hubbleKeys["hubble_drop_total"] = drop
+}
```

Comment:
given this init is var instantiation, we could move this to the var() block where it's declared and avoid the additional init

```suggestion

var (
    flow       = []string{"source", "destination"}
    hubbleKeys = map[string][]string{
		    "hubble_flows_processed_total": flow,
             ...
    }
)
```